odl/discr/discr_mappings.py

```diff
@@ -467,219 +482,307 @@ def _call(self, x, out=None):
+            `IntervalProd`.
+        grid :  `TensorGrid`
+            The grid on which to evaluate. Must be contained in
+            the domain of the function set.
+        dspace : `FnBase`
+            Data space providing containers for the values of a
+            discretized object. Its size must be equal to the
+            total number of grid points. Its field must be the same
+            as that of the function space.
+        schemes : `str` or `list` of `str`
+            Indicates which interpolation scheme to use for which axis.
+            A single string is interpreted as a global scheme for all
+            axes.
+        nn_variants : `str` or `list` of `str`, optional
+            Which variant ('left' or 'right') to use in nearest neighbor
+            interpolation for which axis. A single string is interpreted
+            as a global variant for all axes.
+            This option has no effect for schemes other than nearest
+            neighbor.
+        order : {'C', 'F'}, optional
+            Ordering of the values in the flat data arrays. 'C'
+            means the first grid axis varies slowest, the last fastest,
+            'F' vice versa.
+        """
+        if not isinstance(ip_fspace, FunctionSpace):
+            raise TypeError('function space {!r} is not a `FunctionSpace` '
+                            'instance.'.format(ip_fspace))
+        if not isinstance(ip_fspace.domain, IntervalProd):
+            raise TypeError('function space domain {!r} is not an '
+                            '`IntervalProd` instance.'.format(ip_fspace))
+
+        FunctionSetMapping.__init__(self, 'extension', ip_fspace, grid, dspace,
+                                    order, linear=True)
+
+        try:
+            schemes_ = str(schemes + '').lower()  # pythonic string check
+            schemes_ = [schemes_] * self.grid.ndim
+        except TypeError:
+            schemes_ = [str(scm).lower() for scm in schemes]
+
+        if nn_variants is None:
+            variants_ = ['left'] * self.grid.ndim
+        else:
+            try:
+                variants_ = str(nn_variants + '').lower()  # pythonic str check
+                variants_ = [variants_] * self.grid.ndim
+            except TypeError:
+                variants_ = [str(var).lower() for var in nn_variants]
+
+        for i, (scm, var) in enumerate(zip(schemes, variants_)):
+            if scm not in SUPPORTED_INTERP_SCHEMES:
+                raise ValueError("Interpolation scheme '{}' at index {} not "
+                                 "understood.".format(scm, i))
+            if scm == 'nearest' and var not in ('left', 'right'):
+                raise ValueError("Nearest neighbor variant '{}' at index {} "
+                                 "not understood.".format(var, i))
+
+        self._schemes = schemes_
+        self._nn_variants = variants_
+
+    @property
+    def schemes(self):
+        """List of interpolation schemes, one for each axis."""
```

Comment:
Second thought, not sure about this. I rather return the same type regardless of input during init. Otherwise, each time we use this value for iteration, we need to differentiate.